### ISSUE TITLE
Regarding the input file format for torchmd-net training

### ISSUE THREAD

**Issue by SoumyaCYZ** [compsciencelab/amaro#2]
Respected Sir,
                      Hope you are fine and doing well. Currently, I'm learning amaro. I'm using alanine dipeptide for this purpose. I performed 500 ns equilibrium simulation at 300 and 320 K temperatures in GROMACS software. I have computed the force acting on each atom and the non-hydrogen heavy atom coordinates. Should merge the trajectory of 300 and 320 K temperatures before training. What data should I store in the DF5 file (specifically row and columns)?
Regards
Soumya

**Comment by AntonioMirarchi** [comment]
Hi,
from my experience, while the PMF is known to be temperature-dependent, including multiple temperatures can improve the learned representation and enhance generalization. You might also experiment with different percentage compositions when merging datasets to assess the impact on model performance.

Regarding the HDF5 content, it depends on the amount of data. In your case, an [HDF5 dataset](https://github.com/torchmd/torchmd-net/blob/main/torchmdnet/datasets/hdf.py) format should be enough. The content should include `pos`, `forces`, and `types`.  

Alternatively, if you want to use the mdCATH dataset format, you need to generate an `mdcath_source` file along with the corresponding mdCATH HDF5 files. Check the tutorials [here](https://github.com/compsciencelab/mdCATH/tree/main/user) for a better understanding of the required content.

**Comment by SoumyaCYZ** [comment]
Thanks @AntonioMirarchi sir, for the clarification. I am studying the conformational change of Villin-1 in the presence of bioincompatible ionic liquid through ML-CG. For peptides, I have selected only noh heavy backbone atoms. I have used the amber force field for my simulations. I'm facing problems in generating psf file from pdb file using " torchmd_cg.utils.psfwriter import pdb2psf_CA" because torchmd-net 's code selects only c-alpha atoms. Can you suggest some tools to convert pdb to psf file of noh heavy backbone atoms? 
Apart from this, I'm facing problems in computing delta-forces for training in torchmd-net.  For the generation of delta-forces, I need Forcefield.yaml file. What information I should store in the Forcefield.yaml file?  It'd be a great help if could you share or suggest some code for generating a Forcefield.yaml file.

**Comment by AntonioMirarchi** [comment]
I think that you’re mixing up projects. This is the AMARO repo, which doesn’t use priors so there’s no need for any delta forces. However,  if you want to use `torchmd_cg.utils.psfwriter` for noh-atoms, you’ll need to modify it by applying` mol.filter('noh') `and defining a new mapping in `torchmd_cg.utils.mappings` specific to your case. I can't help with the Forcefield.yaml file, sorry.

**Comment by SoumyaCYZ** [comment]
Thanks, sir for the clarification. I want to use your AMARO method.  sorry for bothering you. I was a bit confused with the torched-cg and other works. I have saved the coordinates and forces in .npy format. Then I should make embeddings.npy file "get_emb.py" file. However, I didn't understand the role of the code present in scaleup_validation and scripts.  Then I think I can train the model in torchmd-net and get CG potential right? It'll be nice if you clarify my doubts.

**Comment by SoumyaCYZ** [comment]
Sir, in train.yaml file instead of source_file: mdcath_noh_source.h5 could I directly use forces.npy and coordinates.npy file from reference all-atom simulation?

**Comment by AntonioMirarchi** [comment]
The `scaleup_validation` and `scripts` contain code to reproduce the analysis presented in the paper. However, if you want to use directly the npys for forces and coords consider the [custom dataset](https://github.com/torchmd/torchmd-net/blob/main/torchmdnet/datasets/custom.py) in TorchMD-Net.

**Comment by SoumyaCYZ** [comment]
Thanks sir, for your suggestion. In custom dataset can I directly use actual forces from atomistic simulation (forces acting on selected backbone heavy atoms)? However, in protein thermodynamics  (nat comm) paper, they have used detlaforces.npy file.  Basically, in the force matching algorithm, they reduce the gap between SCHNET predicted forces (based on cartesian coordinates and prior forcefield) and true atomistic forces. I have used the hypermeters for chignolin, and I'm getting some crazy numbers with each epoch. Is there any rule of thumb to wisely tune the hyperparameters? For alanine-dipeptide (C-N-CA-C-N atoms), I have used the following hyperparameters
activation: silu
aggr: add
atom_filter: -1
batch_size: 16
coord_files: null
cutoff_lower: 0.0
cutoff_upper: 5.0
dataset: Custom
coord_files: "31jan_coords.npy"
embed_files: "31jan_ca_embeddings.npy"
force_files: "31jan_ca_deltaforces.npy"
derivative: false
early_stopping_patience: 150
ema_alpha_neg_dy: 1.0
ema_alpha_y: 1.0
embed_files: null
embedding_dimension: 256
energy_files: null
equivariance_invariance_group: O(3)
y_weight: 1.0
force_files: null
neg_dy_weight: 0.0
gradient_clipping: 40
inference_batch_size: 128
load_model: null
log_dir: /home/chemistry/phd/cyz218385/scratch/aladi_wat/aladi_300k
lr: 0.0001
lr_factor: 0.8
lr_min: 1.0e-07
lr_patience: 15
lr_warmup_steps: 1000
max_num_neighbors: 64
max_z: 128
model: graph-network
ngpus: -1
num_epochs: 3000
num_layers: 3
num_nodes: 1
num_rbf: 64
num_workers: 6
output_model: Scalar
precision: 32
prior_model: Atomref
rbf_type: expnorm
redirect: false
reduce_op: add
save_interval: 10
seed: 1
splits: null
standardize: false
test_interval: 20
test_size: null
train_size: 110000
trainable_rbf: false
val_size: 10000
weight_decay: 0.0
charge: false
spin: false

**Comment by SoumyaCYZ** [comment]
![Image](https://github.com/user-attachments/assets/a112dcc9-d462-4131-8de7-5138c0878977)
@AntonioMirarchi sir, I didn't understand whether the learning curve is correct or not because it starts from high number 1.7*e25. I got this learning curve when I used the following hyperparameters
activation: silu
attn_activation: silu
batch_size: 8
inference_batch_size: 9
dataset: Custom
coord_files: "31jan_coords.npy"
embed_files: "31jan_ca_embeddings.npy"
force_files: "31jan_ca_deltaforces.npy"
cutoff_upper: 5.0
cutoff_lower: 0.0
derivative: true
early_stopping_patience: 30
embedding_dimension: 128
distance_influence: both
lr: 0.0003
lr_factor: 0.8
lr_min: 1.0e-06
#lr_metric: val_total_mse_loss
lr_patience: 10
lr_warmup_steps: 0
log_dir: /scratch/chemistry/phd/cyz218385/aladi_wat/aladi_300k
model: graph-network
neighbor_embedding: false
ngpus: -1
num_epochs: 100
num_heads: 8
num_layers: 1
num_nodes: 1
num_rbf: 32
#num_workers: 8
rbf_type: expnorm
save_interval: 5
seed: 42
test_interval: -1
test_size: 0.1
train_size: null
trainable_rbf: false
val_size: 0.05
trainable_rbf: true
weight_decay: 0.0
I picked up a ckpt file at epoch 40; then I tried to run a simulation using " torchmd-train --conf simulate.yaml" torchmd-net is running. Is there any specific command for performing simulation? It will be a great help if you clear my doubts.
